crates/dyn-abi/src/ext/event.rs

```diff
@@ -0,0 +1,301 @@
+                    "
+                    0000000000000000000000000000000000000000000000000000000000000003
+                    0000000000000000000000002222222222222222222222222222222222222222
+                "
+                ),
+                false,
+            )
+            .unwrap();
+
+        assert_eq!(
+            result,
+            [
+                DynSolValue::Int(
+                    Signed::from_be_bytes(hex!(
+                        "0000000000000000000000000000000000000000000000000000000000000003"
+                    )),
+                    256
+                ),
+                DynSolValue::Int(
+                    Signed::from_be_bytes(hex!(
+                        "0000000000000000000000000000000000000000000000000000000000000002"
+                    )),
+                    256
+                ),
+                DynSolValue::Address(address!("2222222222222222222222222222222222222222")),
+                DynSolValue::Address(address!("1111111111111111111111111111111111111111")),
+                DynSolValue::FixedBytes(
+                    b256!("00000000000000000aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+                    32
+                ),
+                DynSolValue::FixedBytes(
+                    b256!("00000000000000000bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"),
+                    32
+                ),
+                DynSolValue::FixedBytes(
+                    b256!("00000000000000000ccccccccccccccccccccccccccccccccccccccccccccccc"),
+                    32
+                ),
+            ]
+        );
+    }
+
+    #[test]
+    fn parse_log_whole() {
+        let correct_event = Event {
+            name: "Test".into(),
+            inputs: vec![
+                EventParam {
+                    ty: "(address,address)".into(),
+                    indexed: false,
+                    ..Default::default()
+                },
+                EventParam {
+                    ty: "address".into(),
+                    indexed: true,
+                    ..Default::default()
+                },
+            ],
+            anonymous: false,
+        };
+        // swap indexed params
+        let mut wrong_event = correct_event.clone();
+        wrong_event.inputs[0].indexed = true;
+        wrong_event.inputs[1].indexed = false;
+
+        let log = Log::new(
+            vec![
+                b256!("cf74b4e62f836eeedcd6f92120ffb5afea90e6fa490d36f8b81075e2a7de0cf7"),
+                b256!("0000000000000000000000000000000000000000000000000000000000012321"),
+            ],
+            bytes!(
+                "
+			0000000000000000000000000000000000000000000000000000000000012345
+			0000000000000000000000000000000000000000000000000000000000054321
+			"
+            ),
+        );
+
+        wrong_event.decode_log_object(&log, false).unwrap();
+        // TODO: How do we verify here?
+        // wrong_event.decode_log_object(&log, true).unwrap_err();
```

Comment:
what do we want to check?

Reply:
I guess that decoding 2 words into 1 (non indexed address) should fail? But currently doesn't, I'm not really sure how to fix this and it's out of scope

Reply:
[`ethabi::decode_whole`](https://docs.rs/ethabi/latest/ethabi/fn.decode_whole.html) says it "Fails, if some data left to decode" so I guess that would be enough to fix this